Source/iop/Iop_Sysclib.cpp

```diff
@@ -241,6 +248,16 @@ void CSysclib::Invoke(CMIPS& context, unsigned int functionId)
 			context.m_State.nGPR[CMIPS::V0].nD0 = context.m_State.nGPR[CMIPS::A0].nV0;
 		}
 		break;
+	case 42:
+		// int vsprintf (char * s, const char * format, va_list arg );
+		{
+			context.m_State.nGPR[CMIPS::V0].nD0 = static_cast<int>(__vsprintf(
+				reinterpret_cast<char*>(&m_ram[context.m_State.nGPR[CMIPS::A0].nV0]),
+				reinterpret_cast<const char*>(&m_ram[context.m_State.nGPR[CMIPS::A1].nV0]),
+				reinterpret_cast<va_list>(&m_ram[context.m_State.nGPR[CMIPS::A2].nV0])

```

Comment:
Actually you can see this use pattern in the implementation of vxprintf in the file you link to below.